pkg/storage/stores/shipper/indexshipper/tsdb/single_file_index.go

```diff
@@ -362,6 +359,24 @@ func (i *TSDBIndex) Volume(
 	seriesLabels := labels.Labels(make([]labels.Label, 0, len(labelsToMatch)))
 
 	aggregateBySeries := seriesvolume.AggregateBySeries(aggregateBy) || aggregateBy == ""
+	var by map[string]struct{}
+	if !includeAll && (aggregateBySeries || len(targetLabels) > 0) {
+		by = make(map[string]struct{}, len(labelsToMatch))
+		for k := range labelsToMatch {
+			by[k] = struct{}{}
+		}
+		if len(labelsToMatch) > 0 {
```

Comment:
why are we iterating over `labelsToMatch` twice?

Reply:
Yeah this was supposed to be targetLabels